File: examples/java/kora-java-scheduling-quartz/src/main/java/io/koraframework/example/scheduling/quartz/TriggerScheduler.java
package io.koraframework.example.scheduling.quartz;

import io.koraframework.common.annotation.Component;
import io.koraframework.scheduling.quartz.ScheduleWithTrigger;

@Component
public final class TriggerScheduler {

    private int state = 0;

    /**
     * @see Application#myTrigger()
     */
    @ScheduleWithTrigger(TriggerScheduler.class)
    void schedule() {
        state++;
    }

    public int getState() {
        return state;
    }
}

File: examples/kotlin/kora-kotlin-scheduling-quartz/src/main/kotlin/io/koraframework/kotlin/example/scheduling/quartz/TriggerScheduler.kt
package io.koraframework.kotlin.example.scheduling.quartz

import io.koraframework.common.annotation.Component
import io.koraframework.scheduling.quartz.ScheduleWithCron
import io.koraframework.scheduling.quartz.ScheduleWithTrigger

@Component
class TriggerScheduler {
    var state = 0
        private set

    @ScheduleWithTrigger(TriggerScheduler::class)
    fun schedule() {
        state++
    }
}
